src/procfs.cpp

```diff
@@ -137,6 +140,50 @@ uptime procfs::get_uptime() const
     return parsers::parse_uptime_line(line);
 }
 
+sys_stat procfs::get_stat() const
```

Comment:
So there was actually already a parser for files like that.
It's called `file_parser` and it's defined in `parsers.hpp`.
It goes over the file line by line, splits the line into two and then calls the right parser from an unordered map of parsers.
This is the way we parse the `/proc/<pid>/status` file (The `status.cpp` file is a great reference).
The only thing that you should probably change is to get this class accept the DELIM as a parameter instead of using the hard coded ':'.

Reply:
I guess only this method should be replaced with the parser instance. The per line parsers are still going to be required obviously.

Reply:
You are right, except the number of CPUs is not constant : cpu1 cpu2 ... So using a map might be problematic here, unless I add some specific changes (for example if we have cpuX, remove the X and find cpu). Maybe we can solve that by getting a function as a parameter which manipulates the keys and only then calls the suitable function from the map. What do you think?

Reply:
Hmm, that's a good point...
I like your idea. I think that we can pass that key "permutation" function when constructing the parser.
There are a few things to consider though:
1. The function should probably return a copy of the key? For example, return `cpuX` from `cpu1` because we still need to pass the actual key to the line parsers. I guess we'll need the value `1` to add that to the correct index in the cpus `vector`. I think that we cannot assume correct alphanumeric order with more than 10 cpus (possibly ordered: 1, 10, 11, ..., 2, 3, ....)
2. Today we don't even pass the key to the parser, this should probably change as well?
3. On the other hand, we don't want to copy all the keys when there's no good reason

I guess this is going to be challenging. Best way is probably to try and see how it turns out, but I believe you can find the right solution :)
If it makes things too complicated, we can re-consider using the current solution, so make sure you stash it somewhere.

Reply:
1. As I see it - we need another function to provide another level of translation. For example in our case:
cpuX -> cpu

So all the cpu lines will be parsed in the same manner.

2. I don't think we need to care about the order of the CPUs, we should add them in the same order as the lines order.